pyomo/gdp/plugins/bigm.py

```diff
@@ -175,76 +195,51 @@ def _apply_to_impl(self, instance, **kwds):
+        for t in targets:
+            # check that t is in fact a child of instance
+            if not is_child_of(parent=instance, child=t,
+                               knownBlocks=knownBlocks):
+                raise GDP_Error("Target %s is not a component on instance %s!"
+                                % (t.name, instance.name))
+            elif t.type() is Disjunction:
                 if t.parent_component() is t:
-                    self._transformDisjunction(t, transBlock, bigM)
+                    self._transform_disjunction(t, bigM)
                 else:
-                    self._transformDisjunctionData(
-                        t, transBlock, bigM, t.index())
+                    self._transform_disjunctionData( t, bigM, t.index())
             elif t.type() in (Block, Disjunct):
                 if t.parent_component() is t:
-                    self._transformBlock(t, transBlock, bigM)
+                    self._transform_block(t, bigM)
                 else:
-                    self._transformBlockData(t, transBlock, bigM)
+                    self._transform_blockData(t, bigM)
             else:
                 raise GDP_Error(
                     "Target %s was not a Block, Disjunct, or Disjunction. "
                     "It was of type %s and can't be transformed."
                     % (t.name, type(t)))
-        # Go through our dictionary of indexed things and deactivate
-        # the containers that don't have any active guys inside of
-        # them. So the invalid component logic will tell us if we
-        # missed something getting transformed.
-        for obj in transBlock.disjContainers:
-            if not obj.active:
-                continue
-            for i in obj:
-                if obj[i].active:
-                    break
-            else:
-                # HACK due to active flag implementation.
-                #
-                # Ideally we would not have to do any of this (an
-                # ActiveIndexedComponent would get its active status by
-                # querring the active status of all the contained Data
-                # objects).  As a fallback, we would like to call:
-                #
-                #    obj._deactivate_without_fixing_indicator()
-                #
-                # However, the sreaightforward implementation of that
-                # method would have unintended side effects (fixing the
-                # contained _DisjunctData's indicator_vars!) due to our
-                # class hierarchy.  Instead, we will directly call the
-                # relevant base class (safe-ish since we are verifying
-                # that all the contained _DisjunctionData are
-                # deactivated directly above).
-                ActiveComponent.deactivate(obj)
+
+        # issue warnings about anything that was in the bigM args dict that we
+        # didn't use
+        if not bigM is None and len(bigM) > len(used_args):
```

Comment:
PEP8 style: `if bigM is not None`

Reply:
Also, why not actually do a set comparison (e.g., `set(bigM) - set(used_args)`)?  Because of line 713 (`used_args[block] = val`), are we guaranteed that just comparing the number of entries is sufficient (block comes from arg_list and not bigMargs)?

Reply:
Oh, yeah, the set comparison is better. We have to use ComponentSet though in case someone messes up and gives us something unhashable (Var). (I think number of entries would be sufficient because we never add anything to arg_list that is not in bigMargs, and we add all the Blocks that we could possible need. But since we do the set comparison to get the warning message, might as well do it initially.)